Kernel/Arch/x86/common/Processor.cpp

```diff
@@ -236,12 +236,13 @@ UNMAP_AFTER_INIT void Processor::cpu_setup()
     MSR star_msr(MSR_STAR);
     star_msr.set(star);
 
-    // Write the syscall entry point to the LSTAR MSR, and write the SFMASK MSR to clear rflags upon entry.
-    // The userspace rflags will be preserved in r11.
+    // Write the syscall entry point to the LSTAR MSR, and write the SFMASK MSR to clear rflags upon entry. The value
+    // set here clears most of rflags, but keeps the reserved and virtualization bits intact. The userspace rflags
+    // will be preserved in r11.
     MSR lstar_msr(MSR_LSTAR);
     MSR sfmask_msr(MSR_SFMASK);
     lstar_msr.set(reinterpret_cast<u64>(&syscall_entry));
-    sfmask_msr.set(~0x2);
+    sfmask_msr.set(0x257fd5u);
```

Comment:
This looks good (although it might be nicer to extract this magical number to a constexpr constant), though the ordering of the comment and writes is a bit weird, can we separate the two into 2 separate comments and 2 separate MSR writes just for clarity and consistency with the rest of the writes?